internal/js/modules/k6/browser/tests/page_test.go

```diff
@@ -2182,3 +2183,295 @@ func TestPageOnMetric(t *testing.T) {
+				headers: request.headers(),
+				headersArray: await request.headersArray(),
+				isNavigationRequest: request.isNavigationRequest(),
+				method: request.method(),
+				postData: request.postData(),
+				postDataBuffer: request.postDataBuffer() ? String.fromCharCode.apply(null, new Uint8Array(request.postDataBuffer())) : null,
+				resourceType: request.resourceType(),
+				// Ignoring response for now since it is not reliable as we don't explicitly wait for the request to finish.
+				// response: await request.response(),
+				size: request.size(),
+				// Ignoring timing for now since it is not reliable as we don't explicitly wait for the request to finish.
+				// timing: request.timing(),
+				url: request.url()
+			});
+		});
+
+		await page.goto('%s', {waitUntil: 'networkidle'});
+
+		await page.close();
+
+		return JSON.stringify(returnValue, null, 2);
+	`, tb.url("/home"))
+	assert.NoError(t, err)
+
+	got := k6test.ToPromise(t, gv)
+
+	// Convert the result to a string and then to a slice of requests.
+	var requests []request
+	err = json.Unmarshal([]byte(got.Result().String()), &requests)
+	require.NoError(t, err)
+
+	for i := range requests {
+		// Normalize any port numbers in the string values to :8080
+		if requests[i].URL != "" {
+			requests[i].URL = regexp.MustCompile(`:\d+`).ReplaceAllString(requests[i].URL, ":8080")
+		}
+		if requests[i].FrameURL != "" {
+			requests[i].FrameURL = regexp.MustCompile(`:\d+`).ReplaceAllString(requests[i].FrameURL, ":8080")
+		}
+		for k, v := range requests[i].AllHeaders {
+			requests[i].AllHeaders[k] = regexp.MustCompile(`:\d+`).ReplaceAllString(v, ":8080")
+		}
+		for k, v := range requests[i].Headers {
+			requests[i].Headers[k] = regexp.MustCompile(`:\d+`).ReplaceAllString(v, ":8080")
+		}
+		for k, header := range requests[i].HeadersArray {
+			if header["value"] != "" {
+				requests[i].HeadersArray[k]["value"] = regexp.MustCompile(`:\d+`).ReplaceAllString(header["value"], ":8080")
+			}
+		}
+	}
+
+	expected := []request{
+		{
+			AllHeaders: map[string]string{
+				"accept-language":           "en-US",
+				"upgrade-insecure-requests": "1",
+				"user-agent":                "some-user-agent",
+			},
+			FrameURL:             "about:blank",
+			AcceptLanguageHeader: "en-US",
+			Headers: map[string]string{
+				"Accept-Language":           "en-US",
+				"Upgrade-Insecure-Requests": "1",
+				"User-Agent":                "some-user-agent",
+			},
+			HeadersArray: []map[string]string{
+				{"name": "Upgrade-Insecure-Requests", "value": "1"},
+				{"name": "User-Agent", "value": "some-user-agent"},
+				{"name": "Accept-Language", "value": "en-US"},
+			},
+			IsNavigationRequest: true,
+			Method:              "GET",
+			PostData:            "",
+			PostDataBuffer:      "",
+			ResourceType:        "Document",
+			Size: map[string]int{
+				"body":    0,
+				"headers": 103,
+			},
+			URL: "http://127.0.0.1:8080/home",
```

Comment:
Instead of a static URL that we need to transform (right above the `expected` variable), we can also use `tb.url` to get the server's URL:

```suggestion
			URL: tb.url("/home"),
```

Reply:
Oh, nice suggestion!